lib/mix/lib/mix/tasks/compile.make.ex

```diff
@@ -0,0 +1,98 @@
+defmodule Mix.Tasks.Compile.Make do
+  use Mix.Task
+
+  @shortdoc "Runs `make` in the current project"
+
+  @moduledoc """
+  Runs `make` in the current project.
+
+  This task runs `make` in the current project; any output coming from `make` is
+  printed in real-time on stdout. `make` will be called without specifying a
+  Makefile, so there has to be a `Makefile` in the current working directory.
+
+  ## Configuration
+
+    * `:make_makefile` - it's a binary. It's the Makefile to use. Defaults to
+      `"Makefile"` for Unix systems and `"Makefile.win"` for Windows systems.
+
+    * `:make_targets` - it's a list of binaries. It's the list of Make targets
+      that should be run. Defaults to `[]`, meaning `make` will run the first
+      target.
+
+    * `:make_cwd` - it's a binary. It's the directory where `make` will be run,
+      relative to the root of the project.
+
+    * `:make_error_message` - it's a binary. It's a custom error message that
+      can be used to give instructions as of how to fix the error (e.g., it can
+      be used to suggest installing `gcc` if you're compiling a C dependency).
+
+  """
+
+  @spec run([binary]) :: :ok | no_return
+  def run(_args) do
+    config = Mix.Project.config()
+    build(config)
+    Mix.Project.build_structure
+    :ok
+  end
+
+  defp build(config) do
+    makefile  = Keyword.get(config, :make_makefile, :default)
```

Comment:
Should we also expose `make_executable`? The default is the OS lookup we perform. :)